### PR TITLE
refactor($errorCatch): allow to log detailed error message

### PR DESCRIPTION
**Pull request by ulivz**
# Overview

- Add detailed error log
- in `watch` mode, bili return `watchers` instead of `undefined`
- Increase cov from `71%` to `97%+` (In local is 99.36%...).

# More

When I use `bili` to bundle an module with complicated dependencies, it throw error:

image

But in that case I can't get my error detail, so I can't know why it's failing

So I open this PR, now the error log will be like this which it's human to let me know the error primary cause:

image

> Digression: I have to blow a wave of `bili`, so fucking good to use 👏

### CODE REVIEW

File: src/get-config.js [hunk 1]
```diff
@@ -20,7 +20,6 @@ function readInPkg(file) {
         pkg: {}
       }
     }
-    throw err
```
Comment (egoist) [inline src/get-config.js]: hmm why removing this?

File: src/utils.js [hunk 2]
```diff
@@ -13,5 +13,7 @@ export function handleRollupError(error) {
     console.log(chalk.dim(`Location: ${error.id}`))
   }
   console.error(error.snippet)
+  console.log()
+  console.log(error)
```
Comment (egoist) [inline src/utils.js]: maybe only log error when there's no `error.snippet`?

**Comment by ulivz** [comment]
Hey, dear @egoist, Since I am familiar with the whole design logic and code of bili, I would like to apply to become the main maintainer of bili, I will be responsible for each line of code, it seems to be self-recommendation, do not know whether or not, very honored if I can get this opportunity.

File: test/index.test.js [hunk 3]
```diff
@@ -19,22 +32,79 @@ afterAll(() => {
...
+      replace: {
+        __VERSION__: '0.0.0'
+      },
+      write: false
+    })
+    expect(cjs.code).toMatchSnapshot()
+  })
+
+  test('should throw error when "format" is neither string nor array', () => {
+    return Promisify(bili, {
```
Comment (egoist) [inline test/index.test.js]: hmm I guess it already returns a Promise?

File: test/index.test.js [hunk 4]
```diff
@@ -138,6 +208,157 @@ test('generate all bundles', async () => {
...
+  const pkgConfig = getConfig(cwd('fixtures/bili.config.js'))
+  console.log(pkgConfig)
+  expect(pkgConfig.name).toBe('EGOIST')
+  expect(typeof pkgConfig.pkg).toBe('object')
+  expect(JSON.stringify(pkgConfig.pkg)).toBe('{}')
+})
+
+describe('get rollup options', () => {
+  test('should get correct ouput file name', () => {
+    return Promisify(getRollupOptions, {
```
Comment (egoist) [inline test/index.test.js]: why promisify `getRollupOptions`  😅

**Comment by egoist** [comment]
hmm why is this closed 😅

**Pull request closed without merging**